### ISSUE TITLE
Service Provider boot run before app Locale change

### ISSUE THREAD

**Issue by fh32000** [pishran/nova-rtl-theme#17]
I have Middleware to change  app Locale but ThemeServiceProvider boot run before my  Middleware

`<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\Log;
````

class Localization
{
    /**
     * Handle an incoming request.
     *
     * @param \Illuminate\Http\Request $request
     * @param \Closure $next
     * @return mixed
     */
    public function handle($request, Closure $next)
    {
        if (session()->has('locale')) {
            app()->setLocale(session()->get('locale'));
        } elseif ($request->has('locale')) {
            app()->setLocale(request('locale'));
        } elseif (Auth::check()) {

            Log::debug('Auth::check');

            $user = request()->user();
            $prefix = $user->id;
            $locale_key = $prefix . ".locale";
            $locale_val = cache()->get($locale_key);
            if (!is_null($locale_val)) {
                app()->setLocale($locale_val);
            }

        }

        Log::info( app()->getLocale());
        return $next($request);
    }
}
`
```


`    /**
     * The application's route middleware groups.
     *
     * @var array
     */
    protected $middlewareGroups = [
        'web' => [
            \App\Http\Middleware\EncryptCookies::class,
            \Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse::class,
            \Illuminate\Session\Middleware\StartSession::class,
            // \Illuminate\Session\Middleware\AuthenticateSession::class,
            \App\Http\Middleware\Localization::class,
           ...
        ],
        `

```

<?php

namespace Pishran\NovaRtlTheme;

use Illuminate\Support\Facades\Log;
use Illuminate\Support\ServiceProvider;
use Laravel\Nova\Events\ServingNova;
use Laravel\Nova\Nova;

class ThemeServiceProvider extends ServiceProvider
{
    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot()
    {
        Log::debug('boot');
        Log::debug(app()->getLocale());
        if (in_array(app()->getLocale(), config('nova-rtl-theme.locales', []))) {
            Nova::serving(function (ServingNova $event) {
                Nova::provideToScript([
                    'nova_rtl_theme' => [
                        'stylesheet' => config('nova-rtl-theme.stylesheet'),
                        'font_family' => config('nova-rtl-theme.font-family'),
                    ],
                ]);
                
                
                

                Nova::style('nova-rtl-theme', __DIR__ . '/../resources/css/theme.css');

                Nova::script('nova-rtl-theme', __DIR__ . '/../resources/js/theme.js');
            });
        }

        $this->publishes([
            __DIR__ . '/../config/nova-rtl-theme.php' => config_path('nova-rtl-theme.php'),
        ], 'nova-rtl-theme');
    }

    /**
     * Register any application services.
     *
     * @return void
     */
    public function register()
    {
        $this->mergeConfigFrom(
            __DIR__ . '/../config/nova-rtl-theme.php', 'nova-rtl-theme'
        );
    }
}
```



**Log:**
```
[2021-02-22 15:10:21] local.DEBUG: boot  
[2021-02-22 15:10:21] local.DEBUG: en  
[2021-02-22 15:10:21] local.DEBUG: Auth::check  
```

**Comment by fh32000** [comment]
@omarhen @sepehrr @FaridAghili @SadeghPM @

**Comment by fh32000** [comment]
it work affer this change in your code 

```
`        Nova::serving(function (ServingNova $event) {
            if (in_array(app()->getLocale(), config('nova-rtl-theme.locales', []))) {
                Nova::provideToScript([
                    'nova_rtl_theme' => [
                        'stylesheet' => config('nova-rtl-theme.stylesheet'),
                        'font_family' => config('nova-rtl-theme.font-family'),
                    ],
                ]);

                Nova::style('nova-rtl-theme', __DIR__ . '/../resources/css/theme.css');

                Nova::script('nova-rtl-theme', __DIR__ . '/../resources/js/theme.js');
            }
        });`
```

**Comment by fh32000** [comment]
#18

**Comment by FaridAghili** [comment]
Thanks
Fixed in v1.5.9
